libsql/internal/hrana/batch_result.go

```diff
@@ -1,6 +1,51 @@
 package hrana
 
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
 type BatchResult struct {
-	StepResults []*StmtResult `json:"step_results"`
-	StepErrors  []*Error      `json:"step_errors"`
+	StepResults      []*StmtResult `json:"step_results"`
+	StepErrors       []*Error      `json:"step_errors"`
+	ReplicationIndex *uint64       `json:"replication_index"`
```

Comment:
Do we need omitempty here?

Reply:
It is not required. `omitempty` affects when the data is being encoded. In this case, we don't marhsal `BatchResult` anywhere and only decode it from the server. That's why I didn't add it.